### PR TITLE
[SL-UP] Skip the OpenThread API call for WiFi builds

### PR DESCRIPTION
**Pull request by selissia**
Build fails for thermostat-rs911x.slcp if the matter_shell component is excluded. To fix skip the OpenThread API call in uart.cpp in case of WiFi builds. 
Verified in a FULL CI build on the matter_extension repo.

### CODE REVIEW

**Review by senthilku — APPROVED** [review]

File: examples/platform/silabs/uart.cpp
Comment (mkardous-silabs) [inline examples/platform/silabs/uart.cpp:416]: Should we instead define gate with `CHIP_DEVICE_CONFIG_THREAD_ENABLE_CLI` since the Ot-Cli should not be enabled?

Comment (selissia) [reply examples/platform/silabs/uart.cpp:416]: Perhaps. But not in this PR which is a hotfix to unblock the release build. The SL_WIFI macro is already used in other places in this file.

Comment (mkardous-silabs) [reply examples/platform/silabs/uart.cpp:416]: I don't see the relevance with a hotfix but i won't block the PR. Can you file a follow up item when the upstream PR is open?

**Review by mkardous-silabs — APPROVED** [review]

File: examples/platform/silabs/uart.cpp
Comment (selissia) [reply examples/platform/silabs/uart.cpp:416]: MATTER-4295 has been created to follow up

**Pull request merged**
